lib/smartcar/utils.rb

```diff
@@ -102,7 +102,7 @@ def process_batch_response(response_body, response_headers)
         attribute_name = convert_path_to_attribute(item['path'])
         aliases = Vehicle::METHODS[attribute_name.to_sym][:aliases]
         # merging the top level request headers and separate headers for each item of batch
-        headers = response_headers.merge(item['headers'])
+        headers = response_headers.merge(item['headers'] || {})
```

Comment:
sneaked a bugfix in here. In case when one of the batch items is an error, the response doesnt have a `headers` so it is `nil` and this ends up throwing an error trying to merge `nil` into hash.